R/utils-hub_tasks_config.R

```diff
@@ -1,39 +1,21 @@
-#' For each task group, get the target_id entries from its target_metadata
-#'
-#' @return a list with one entry for each task group,
-#' where each entry is a character vector of target_ids for that group
+#' Filter task groups from a hub's tasks config to those that contain a target_id.
+#' Additionally, subset the target_metadata to just the entry for the target_id.
 #'
 #' @noRd
-get_target_ids_by_task_group <- function(task_groups) {
-  result <- purrr::map(
+filter_task_groups_to_target <- function(task_groups, target_id) {
+  # For each task group, subset the target_metadata to just the entry for the target_id
+  # If the target_id is not in the task group, the target_metadata will be empty
+  task_groups <- purrr::map(
     task_groups,
     function(task_group) {
```

Comment:
This is a much better second iteration! I would even go so far as to say that you could move this anonymous function into a named function (maybe `trim_to_target_id`).